test/scripts/get-missing-cherry-picks.test.ts

```diff
@@ -1,150 +1,222 @@
 import assert from 'assert';
-import nock from 'nock';
+import {Octokit} from '@octokit/rest';
+import {GetResponseTypeFromEndpointMethod} from '@octokit/types';
+import {spy, when, deepEqual, resetCalls, reset} from 'ts-mockito';
 import {getMissingCommits} from '../../scripts/get-missing-cherry-picks-utils';
 
-const url = 'https://api.github.com';
 describe('cases where cherry-picks are missing', () => {
-  before(() => {
-    nock.disableNetConnect();
-  });
+  const octokit = new Octokit();
+  const spiedRepos = spy(octokit.rest.repos);
```

Comment:
What happens if you spy `octokit` instead of `octokit.rest.repos`? Can that still work? I prefer that cause what if we end up needing a different path of the API for whatever reason and forget to update the tests? They might accidentally pass if the calls go through to a real octokit instance :-o

Reply:
ts-mockito doesn't support deep mocks unfortunately. I guess it's a matter of how much of a blocker this is.